x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/FieldPermissionsCache.java

```diff
@@ -72,26 +72,25 @@ public FieldPermissions getFieldPermissions(FieldPermissionsDefinition fieldPerm
     }
 
     /**
-     * Returns a field permissions object that corresponds to the merging of the given field permissions and caches the instance if one was
-     * not found in the cache.
+     * Returns a field permissions object that corresponds to the union of the given field permissions.
+     * Union means a field is granted if it is granted by any of the FieldPermissions from the given
+     * collection.
+     * The returned instance is cached if one was not found in the cache.
      */
-    FieldPermissions getFieldPermissions(Collection<FieldPermissions> fieldPermissionsCollection) {
+    FieldPermissions union(Collection<FieldPermissions> fieldPermissionsCollection) {
```

Comment:
Rename this method because it should only be used for union field permissions within the same role. It must *not* be used for multiple limiting (intersecting) roles.